marlowe-playground-client/src/Simulation.purs

```diff
@@ -446,61 +458,74 @@ transactionComposer ::
   forall p.
   State ->
   HTML p Action
-transactionComposer state =
-  div [ classes [ ClassName "transaction-composer", ClassName "composer" ] ]
-    [ ul [ class_ (ClassName "participants") ]
-        if (Map.isEmpty possibleActions) then
-          [ text "No valid inputs can be added to the transaction" ]
-        else
-          (actionsForParties possibleActions)
-    , div [ class_ (ClassName "transaction-btns") ]
-        [ ul [ classes [ ClassName "demo-list", aHorizontal ] ]
-            [ li [ classes [ bold, pointer ] ]
-                [ a
-                    [ onClick $ const
-                        $ if hasHistory state then
-                            Just Undo
-                          else
-                            Nothing
-                    , class_ (Classes.disabled $ not isEnabled)
-                    ]
-                    [ text "Undo" ]
-                ]
-            , li [ classes [ bold, pointer ] ]
-                [ a
-                    [ onClick $ const
-                        $ if hasHistory state then
-                            Just ResetSimulator
-                          else
-                            Nothing
-                    , class_ (Classes.disabled $ not isEnabled)
-                    ]
-                    [ text "Reset" ]
-                ]
-            ]
-        ]
-    ]
-  where
-  currentBlock = state ^. (_marloweState <<< _Head <<< _slot)
-
-  isEnabled = isContractValid state
+transactionComposer state
+  | isNothing (state ^. (_marloweState <<< _Head <<< _executionState)) =
```

Comment:
here is a nice usecase for `has` (actually `hasn't`). I think `hasn't (_marloweState <<< _Head <<< _executionState) state)` will work